pkg/workloadmeta/types.go

```diff
@@ -546,16 +555,40 @@ type CollectorEvent struct {
 	Entity Entity
 }
 
-// Event is an event generated by the metadata store, to be broadcasted to
-// subscribers.
+// Event represents a change to an entity.
 type Event struct {
-	Type    EventType
+	// Type gives the type of this event.
+	//
+	// When Type is EventTypeSet, this represents an added or updated entity.
+	// Multiple set events may be sent for a single entity.
+	//
+	// When Type is EventTypeUnset, this represents either a removed entity or
+	// a single source removing its information about an entity.  Multiple
+	// unset events may be sent for a single entity, if multiple sources
+	// provided information about it.
+	Type EventType
+
+	// Sources gives the sources providing the information in the Entity field.
 	Sources []Source
-	Entity  Entity
+
+	// Entity is the entity involved in this event.  For a set event, this may
+	// contain information "merged" from multiple sources, as listed in the
```

Comment:
#10331 removes `Sources` entirely, as we want to discourage clients from trying to guess what's generating events (this whole thing is supposed to be source-agnostic after all), so I wouldn't even mention merging here, as it's just an internal implementation detail.

Reply:
Done.